stdlib/CRC32c/src/CRC32c.jl

```diff
@@ -30,13 +29,14 @@ calling [`take!`](@ref).
 
 For a `String`, note that the result is specific to the UTF-8 encoding
 (a different checksum would be obtained from a different Unicode encoding).
-To checksum an `a::Array` of some other bitstype, you can do `crc32c(reinterpret(UInt8,a))`,
+To checksum an `a::AbstractArray` of some other bitstype without padding,
+you can do `crc32c(vec(reinterpret(UInt8,a)))`,
 but note that the result may be endian-dependent.
 """
 function crc32c end
 
 
-function crc32c(a::DenseBytes, crc::UInt32=0x00000000)
+function crc32c(a::Union{DenseArrayType{UInt8}, AbstractVector{UInt8}}, crc::UInt32=0x00000000)
```

Comment:
Split into two methods: the old `DenseBytes` or whatever it is called now, and a new `AbstractVector{UInt8}` methods (from above).